### PR TITLE
pyflakes F632 Autofix

### PR DESCRIPTION
**Pull request by squiddy**
I haven't done much with Rust yet, so any feedback is appreciated.

### CODE REVIEW

**Comment by charliermarsh** [comment]
Awesome! Thank you for this. Will review tonight or tomorrow AM.

**Comment by squiddy** [comment]
Thank you.

I've since found `SourceGenerator` and it might be better to use that instead of formatting the replacement code manually. I'll give it a try.

**Comment by charliermarsh** [comment]
Yeah using SourceCodeGenerator would be preferable. Even better would be to try and implement the fix with LibCST since then we can preserve more of the original formatting. But it’s a little more involved and SourceCodeGenerator is good too. If you’re interested in the LibCST approach, you could look at how the flake8-comprehensions fixes are implemented.

**Comment by squiddy** [comment]
Agreed, by the looks of it it is more involved, but I've dabbled with LibCST before so I'll give it a try.

**Comment by charliermarsh** [comment]
The strategy isn’t super well-documented but the basic idea is to extract the source code for the expression or statement, parse it with LibCST, modify the tree, then generate source code from the modified tree (and structure the fix as a replacement of the old source code with the new).

**Comment by squiddy** [comment]
What's not clear to me currently is where to put code. I think in a different issue you mentioned you prefer more complex code into plugins. Are those supposed to be just an entrypoint and delegating to implementation in checks/fixes? Some trivial plugins just inline the logic.

**Comment by charliermarsh** [comment]
What you have here looks good / consistent with respect to code organization. “Checks” are typically small in size with no autofix. “Plugins” are typically more complex and/or benefit from direct access to the Checker to orchestrate autofixing. Some plugins call through to logic in “checks” but often I find that overkill.

**Comment by squiddy** [comment]
Amended the last commit to update the README to indicate this rule can be fixed automatically.

File: src/pyflakes/plugins/invalid_literal_comparisons.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    is_constant(expr) && !is_singleton(expr)
+}
+
+/// F632
+pub fn invalid_literal_comparison(
+    checker: &mut Checker,
+    left: &Expr,
+    ops: &[Cmpop],
+    comparators: &[Expr],
+    locator: &SourceCodeLocator,
```
Comment (charliermarsh) [inline src/pyflakes/plugins/invalid_literal_comparisons.rs:39]: Can this be accessed via checker.locator, rather than as an independent argument?

**Comment by charliermarsh** [comment]
Looks great! Am especially impressed if you haven’t done much Rust yet :)

File: src/pyflakes/plugins/invalid_literal_comparisons.rs (hunk 1 above)
Comment (squiddy) [reply src/pyflakes/plugins/invalid_literal_comparisons.rs]: Oh, good catch. Took inspiration from some other plugin here, but checker.locator is nicer. Done.

**Comment by charliermarsh** [comment]
Thank you for this! Pumped to have you as a contributor.

**Pull request merged**